### PR TITLE
feat: add LOC management to SDK.

### PR DESCRIPTION
**Pull request by benoitdevos**
logion-network/logion-internal#480

### CODE REVIEW

File: packages/client/src/LocClient.ts [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+    status: LocRequestStatus;
+    rejectReason?: string;
+    userIdentity?: UserIdentity;
+    closedOn?: string;
+    files: LocFile[];
+    metadata: LocMetadataItem[];
+    links: LocLink[];
+}
+
+export interface LocData {
```
Comment (gdethier) [inline packages/client/src/LocClient.ts:87]: I would move this definition to `Loc.ts` as `LocData` objects are never instantiated in this module.

**Review by gdethier — APPROVED** [review]
Great job! :+1:

File: packages/client/integration/Loc.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    const { alice } = state;
+    const client = state.client.withCurrentAddress(USER_ADDRESS);
+    let locsState = await client.locsState();
+
+    const legalOfficer = new LegalOfficerWorker(alice, state);
+
+    const pendingRequest = await locsState.requestTransactionLoc({
+        legalOfficer: alice,
+        description: "This is a Transaction LOC",
+        userIdentity: {
```
[the remark this answers is not in the document]
Comment (benoitdevos) [reply packages/client/integration/Loc.ts:21]: Correct. But on the other way, this codes allows running this particular test in isolation, for instance with:
```
    fit("requests a Transaction LOC", async () => {
        await requestTransactionLoc(state);
    })
```
which proves to be very handy while developing.

File: packages/client/src/LocClient.ts (hunk 1 above)
Comment (benoitdevos) [reply packages/client/src/LocClient.ts]: Done.

File: packages/client/integration/Loc.ts [hunk 3]
```diff
@@ -0,0 +1,86 @@
+import { State, TEST_LOGION_CLIENT_CONFIG, REQUESTER_ADDRESS } from "./Utils";
+import { LegalOfficer, LogionClient, FullSigner } from "../src";
+import { AxiosFactory } from "../src/AxiosFactory";
+import { UUID, buildApi } from "@logion/node-api";
+import { LocData, LocRequestStatus } from "../src/LocClient";
+import { OpenLoc, PendingRequest } from "../src/Loc";
+
+const USER_ADDRESS = REQUESTER_ADDRESS;
```
[the remark this answers is not in the document]
Comment (benoitdevos) [reply packages/client/integration/Loc.ts]: Done.

**Pull request merged**